api/v1beta1/verticadb_types.go

```diff
@@ -1384,3 +1384,15 @@ func (v *VerticaDB) IsKnownDepotVolumeType() bool {
 	}
 	return false
 }
+
+// getFirstPrimarySubcluster returns the first primary subcluster defined in the vdb
+func (v *VerticaDB) GetFirstPrimarySubcluster() *Subcluster {
+	for i := range v.Spec.Subclusters {
+		sc := &v.Spec.Subclusters[i]
+		if sc.IsPrimary {
+			return sc
+		}
+	}
+	// We should never get here because the webhook prevents a vdb with no primary.
+	return &v.Spec.Subclusters[0]
```

Comment:
Why not return nil?

Reply:
This was copied over from some other code. But I think the rationale was it would avoid a panic in case someone did create a vdb without a primary. I'll change it to nil as I'd imagine other bad things would happen if you didn't have a primary.